Sources/MessagingPush/PushHandling/iOSPushEventListener.swift

```diff
@@ -87,79 +81,82 @@ class iOSPushEventListener: PushEventListener {
     // singleton constructor
     private init() {}
 
-    func onPushAction(_ push: PushNotification, didClickOnPush: Bool) -> Bool {
+    func onPushAction(_ pushAction: PushNotificationAction, completionHandler: @escaping () -> Void) {
         guard let pushClickHandler = pushClickHandler,
               let pushHistory = pushHistory,
               let jsonAdapter = jsonAdapter
         else {
-            return false
+            return
         }
-        logger?.debug("On push action event. push: \(push))")
+        let push = pushAction.push
+        logger?.debug("On push action event. push action: \(pushAction))")
 
         guard !pushHistory.hasHandledPush(pushEvent: .didReceive, pushId: push.pushId, pushDeliveryDate: push.deliveryDate) else {
             // push has already been handled. exit early
-            return true
+            return
         }
 
         guard let parsedPush = CustomerIOParsedPushPayload.parse(pushNotification: push, jsonAdapter: jsonAdapter) else {
             // push did not come from CIO
             // Do not call completionHandler() because push did not come from CIO.
             // Forward the request to all other push click handlers in app to give them a chance to handle it.
 
-            // TODO: move proxy call to NotificationCenterFrameworkAdapterImpl since it's what has knowledge about parameters required to call proxy.
-//            notificationCenterDelegateProxy.pushClicked(push, completionHandler: completionHandler)
+            notificationCenterDelegateProxy.onPushAction(pushAction, completionHandler: completionHandler)
```

Comment:
As said in another comment, it's important that proxy calls are tested in our automated tests. In this PR, we are able to do that!